### PR TITLE
Bring reason-react up to date with reason-js master

### PR DESCRIPTION
**Pull request by sgrove**
A few changes:

1. `ReasonJs.Document.element` is `ReasonJs.Dom.element` now
2. Add a few more fields to `ReactRe.event` type (but not all of them, not sure what the plan is here)
3. Add `allowFullScreen` attribute for elements.

### CODE REVIEW

File: src/reactDOMRe.re [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-external unmountComponentAtNode : ReasonJs.Document.element => unit =
+external unmountComponentAtNode : ReasonJs.Dom.element => unit =
   "unmountComponentAtNode" [@@bs.val] [@@bs.module "react-dom"];
 
-external findDOMNode : ReactRe.reactRef => ReasonJs.Document.element =
+external findDOMNode : ReactRe.reactRef => ReasonJs.Dom.element =
   "findDOMNode" [@@bs.val] [@@bs.module "react-dom"];
 
-external domElementToObj : ReasonJs.Document.element => Js.t {..} = "%identity";
+external domElementToObj : ReasonJs.Dom.element => Js.t {..} = "%identity";
```
Comment (glennsl) [inline src/reactDOMRe.re]: You'll want all these functions to take `ReasonJs.Dom.element_like _`. That way they'll accept any "subtype" of element too.

**Comment by glennsl** [comment]
I think #59 is the plan for events. At least I hope so :)

**Comment by glennsl** [comment]
@sgrove I was also thinking `render` and `unmountComponentAtNode` should take an `element_like` btw.

**Comment by chenglou** [comment]
Same as https://github.com/chenglou/reason-react-example/pull/9#issuecomment-289159419. Thanks.

**Pull request closed without merging**